### PR TITLE
[topgun/k8s] Add test for per-pipeline k8s credentials

### PR DESCRIPTION
**Pull request by taylorsilva**
Fixes https://github.com/concourse/concourse/issues/3808

CC @pivotal-bin-ju

### CODE REVIEW

File: topgun/k8s/kubernetes_creds_mgmt_test.go [hunk 1]
```diff
@@ -88,48 +87,106 @@ var _ = Describe("Kubernetes credential management", func() {
...
+
+		var disableTeamNamspaces = func() {
+			By("creating a namespace made by the user instead of the chart")
+			Run(nil, "kubectl", "create", "namespace", releaseName+"-main")
+			extraArgs = []string{
+				"--set=concourse.web.kubernetes.createTeamNamespaces=false",
+			}
+		}
+
+		var resetExtraArgs = func() {
```
Comment (cirocosta) [inline topgun/k8s/kubernetes_creds_mgmt_test.go]: Hey, do you think that we could get away from this?

I'm afraid that this adds some extra cognitive load for who's reading the tests as now you have to keep track of the fact that `extraArgs` can be reset to zero 🤔  

Wdyt? thanks!

File: topgun/k8s/kubernetes_creds_mgmt_test.go [hunk 2]
```diff
@@ -88,48 +87,106 @@ var _ = Describe("Kubernetes credential management", func() {
...
+			extraArgs = []string{}
+		}
+
+
+		Context("using per-team credentials", func() {
+			normalSecret := "foo"
+			specialKeySecret := "caz"
```
Comment (cirocosta) [inline topgun/k8s/kubernetes_creds_mgmt_test.go]: Hey, wdyt of being more explicit in terms of naming here? E.g., what's "special" about it that "normal" doesn't have? 

Thanks!

File: topgun/k8s/kubernetes_creds_mgmt_test.go [hunk 3]
```diff
@@ -88,48 +87,106 @@ var _ = Describe("Kubernetes credential management", func() {
...
 
+			By("seeing the credentials were resolved by concourse")
 			Expect(string(session.Out.Contents())).To(ContainSubstring("bar"))
 			Expect(string(session.Out.Contents())).To(ContainSubstring("zaz"))
 		}
 
-		Context("using the default namespace created by the chart", func() {
-			It("succeeds", runsBuildWithCredentialsResolved)
-		})
+		var runGetsCachedCredentials = func(normalSecret string, specialKeySecret string) {
```
Comment (cirocosta) [inline topgun/k8s/kubernetes_creds_mgmt_test.go]: Nice! It's pretty cool how much readable it becomes when looking at the context and specs descriptions once everything comes down to those methods.

I was wondering if we could get this out of the `context` block though, just like `createCredentialSecret` is separated. Wdyt?

thanks!

File: topgun/k8s/kubernetes_creds_mgmt_test.go (hunk 2 above)
Comment (taylorsilva) [reply topgun/k8s/kubernetes_creds_mgmt_test.go]: Naming hard. The only thing different about `specialKeySecret` is that instead of it's key in the kv-map isn't `value` it's `baz`.
We'll think on it a bit, we weren't too happy with it either.

File: topgun/k8s/kubernetes_creds_mgmt_test.go (hunk 1 above)
Comment (taylorsilva) [reply topgun/k8s/kubernetes_creds_mgmt_test.go]: We were able to remove this variable from the test!

File: topgun/k8s/kubernetes_creds_mgmt_test.go (hunk 2 above)
Comment (taylorsilva) [reply topgun/k8s/kubernetes_creds_mgmt_test.go]: Renamed these. wdyt?

File: topgun/k8s/kubernetes_creds_mgmt_test.go (hunk 3 above)
Comment (taylorsilva) [reply topgun/k8s/kubernetes_creds_mgmt_test.go]: Moved, looks better 👍
